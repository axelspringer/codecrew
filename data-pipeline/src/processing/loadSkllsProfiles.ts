import { readFileSync } from 'fs'
import { basename } from 'path';
import { SkllsProfile } from "../sklls-cli/profile";
import { getLastCommitDate } from '../util/helper';
const glob = require('glob')

export type SkllsProfiles = { [email: string]: SkllsProfile }
export default loadSkllsProfiles

function loadSkllsProfiles (dataDir: string): SkllsProfiles {
    const profiles = {} as SkllsProfiles
    const filePaths = glob.sync(dataDir + '/**/*.json')
    filePaths.forEach(filePath => {
        const skllsDataRaw = readFileSync(filePath, { encoding: 'utf-8' })
        let skllsData = {} as SkllsProfile
        try {
            skllsData = JSON.parse(skllsDataRaw)
        } catch (error) {
            console.log(`Skipping ${filePath} (cannot parse contents)`)
            return
        }

        const email = basename(filePath, ".json")
        const LastCommit = getLastCommitDate(skllsData)

        profiles[email] = { ...skllsData, LastCommit }
    })
    return profiles
}