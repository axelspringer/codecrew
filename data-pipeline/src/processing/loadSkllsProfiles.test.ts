import { test, expect } from '@jest/globals'
import { resolve } from 'path'
import loadSkllsProfiles from './loadSkllsProfiles'

test('loadSkllsProfiles works as expected', () => {
    const profiles = loadSkllsProfiles(resolve('./src/processing/fixtures/sklls-profiles'))
    expect(Object.values(profiles)).toMatchSnapshot()
})