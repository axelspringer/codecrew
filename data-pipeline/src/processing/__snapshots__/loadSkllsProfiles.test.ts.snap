// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSkllsProfiles works as expected 1`] = `
[
  {
    "Dep": {
      "EcmaScript": {
        "@material-ui/core@4.12.3": {
          "Fri, 29 Apr 2022 09:38:50 +0200": 13,
        },
      },
    },
    "Ext": {
      ".json": {
        "Thu, 28 Apr 2022 16:39:53 +0200": 2,
      },
    },
    "LastCommit": "2022-04-29T07:38:50.000Z",
    "Usernames": [
      "github-username",
    ],
  },
]
`;
